lib/jxl/enc_ac_strategy.cc

```diff
@@ -272,6 +272,60 @@ namespace HWY_NAMESPACE {
 using hwy::HWY_NAMESPACE::ShiftLeft;
 using hwy::HWY_NAMESPACE::ShiftRight;
 
+bool MultiBlockTransformCrossesHorizontalBoundary(
+    const AcStrategyImage& ac_strategy, size_t start_x, size_t y,
+    size_t end_x) {
+  if (start_x >= ac_strategy.xsize() || y >= ac_strategy.ysize()) {
+    return false;
+  }
+  if (y % 8 == 0) {
+    // Nothing crosses 64x64 boundaries, and the memory on the other side
+    // of the 64x64 block may still uninitialized.
+    return false;
+  }
+  end_x = std::min(end_x, ac_strategy.xsize());
+  AcStrategyRow row = ac_strategy.ConstRow(y);
+  for (size_t x = start_x; x < end_x;) {
+    if (row[x].IsMultiblock()) {
```

Comment:
Why is this line here at all?

Reply:
I wasn't aware that first block concept is well-defined for single blocks. Thank you!